sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/implementation/converters/CommunicationIdentifierConverter.java

```diff
@@ -33,30 +34,41 @@ public static CommunicationIdentifier convert(CommunicationIdentifierModel ident
         assertSingleType(identifier);
         String rawId = identifier.getRawId();
 
+        CommunicationIdentifierModelKind kind = identifier.getKind();
+
+        if (kind != null) {
+            if (kind == CommunicationIdentifierModelKind.COMMUNICATION_USER
+                && identifier.getCommunicationUser() != null) {
+                return getCommunicationUserIdentifier(identifier);
+            }
+
+            if (kind == CommunicationIdentifierModelKind.PHONE_NUMBER
+                && identifier.getPhoneNumber() != null) {
+                return getPhoneNumberIdentifier(identifier, rawId);
+            }
+
+            if (kind == CommunicationIdentifierModelKind.MICROSOFT_TEAMS_USER
+                && identifier.getMicrosoftTeamsUser() != null) {
+                return getMicrosoftTeamsUserIdentifier(identifier, rawId);
+            }
+
+            Objects.requireNonNull(rawId);
+            return new UnknownIdentifier(rawId);
+        }
+
         if (identifier.getCommunicationUser() != null) {
-            Objects.requireNonNull(identifier.getCommunicationUser().getId());
-            return new CommunicationUserIdentifier(identifier.getCommunicationUser().getId());
+            return getCommunicationUserIdentifier(identifier);
         }
 
         if (identifier.getPhoneNumber() != null) {
-            PhoneNumberIdentifierModel phoneNumberModel = identifier.getPhoneNumber();
-            Objects.requireNonNull(phoneNumberModel.getValue());
-            return new PhoneNumberIdentifier(phoneNumberModel.getValue()).setRawId(rawId);
+            return getPhoneNumberIdentifier(identifier, rawId);
         }
 
         if (identifier.getMicrosoftTeamsUser() != null) {
-            MicrosoftTeamsUserIdentifierModel teamsUserIdentifierModel = identifier.getMicrosoftTeamsUser();
-            Objects.requireNonNull(teamsUserIdentifierModel.getUserId());
-            Objects.requireNonNull(teamsUserIdentifierModel.getCloud());
-            Objects.requireNonNull(rawId);
-            return new MicrosoftTeamsUserIdentifier(teamsUserIdentifierModel.getUserId(),
-                teamsUserIdentifierModel.isAnonymous())
-                .setRawId(rawId)
-                .setCloudEnvironment(CommunicationCloudEnvironment
-                    .fromString(teamsUserIdentifierModel.getCloud().toString()));
+            return getMicrosoftTeamsUserIdentifier(identifier, rawId);
         }
 
-        Objects.requireNonNull(rawId);
+        Objects.requireNonNull(rawId, "'RawID' of the CommunicationUserIdentifierModel cannot be null.");
         return new UnknownIdentifier(rawId);
     }
```

Comment:
pls refactor according to https://github.com/Azure/azure-sdk-for-net/pull/31627/files#r1000528967

Reply:
Done